Docs/source/Software/Apps.rst

```diff
@@ -260,10 +260,66 @@ MinionsIO Semaphore
 
 Whenever the LTC driver is calling the :term:`SPI <SPI>` function, there's going to be some delay until the SPI transfer is complete. 
 During that delay, we should be executing some other task so once the SPI transfer starts, the system should start 
-waiting for this :term: `semaphore <Semaphore>`. Whenever the SPI transfer is complete, we must signal this semaphore so we need to 
+waiting for this :term:`semaphore <Semaphore>`. Whenever the SPI transfer is complete, we must signal this semaphore so we need to 
 have an :term:`ISR <ISR>` that calls the signal semaphore function whenever a transfer is complete. This ISR is in the BSP SPI module.
 
+Safety Check Semaphore
+======================
+
+When the :ref:`Critical State <Critical state Task: Manthan Upadhyaya>` runs, it checks to see if the Amperes, Voltage, Open Wire,
+and Temperature are safe. This process is done by the Critical State Task using a blocking Pend function on the Safety Check Semaphore.
+Each time a semaphore is posted, its ``cnt`` parameter is incremented. When it is pended, the ``cnt`` parameter is decremented. The task
+uses a blocking pend (wont move past the function until the semaphore is pended ``NUM_FAULT_POINTS`` amount of times).
```

Comment:
I don't think it makes sense to put a general explanation of how semaphores work here. If we want to include this in the docs, maybe we should have a section for explaining general concepts? If not we can link to some external resources like the uCOS-III docs

Reply:
Yea, this is included in the glossary as well.